db/hints/resource_manager.cc

```diff
@@ -142,14 +146,25 @@ void space_watchdog::on_timer() {
                 // not hintable).
                 // If exists - let's take a file update lock so that files are not changed under our feet. Otherwise, simply
                 // continue to enumeration - there is no one to change them.
-                const internal::endpoint_id ep{de.name};
+                const auto maybe_ep = std::invoke([&] () -> std::optional<internal::endpoint_id> {
+                    try {
+                        return internal::endpoint_id{de.name};
+                    } catch (...) {
+                        resource_manager_logger.warn("Encountered a hint directory of invalid name: {}, exception: {}. ",
+                                de.name, std::current_exception());
```

Comment:
AFAIK `space_watchdog::on_timer()` runs every second. This will generate a lot of spam. I don't think we should print it with such high verbosity, let's reduce it to `debug` - the user will know that there are issues with these directories because of the `warn` messages in `manager::start`.

Reply:
That's a good point, thank you